### PR TITLE
chore: Remove last Travis build

### PR DESCRIPTION
**Pull request by coryan**
Make the clang-tidy build perform all the same checks as the last
Travis build. In addition, this build now runs the same checks as build
with the same name in the Cloud Spanner project.

I also removed the Travis configuration file. A future PR (or probably
more than one) will remove the scripts that we used for Travis.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2852)

### CODE REVIEW

File: ci/kokoro/docker/build.sh [hunk 1]
```diff
@@ -287,14 +288,16 @@ exit_status=$?
 echo "Build finished with ${exit_status} exit status $(date)."
 echo "================================================================"
 
+"${PROJECT_ROOT}/ci/kokoro/docker/upload-docs.sh"
+
 "${PROJECT_ROOT}/ci/kokoro/docker/upload-coverage.sh" "${docker_flags[@]}"
 
 echo "================================================================"
-"${PROJECT_ROOT}/ci/travis/dump-logs.sh"
+#"${PROJECT_ROOT}/ci/travis/dump-logs.sh"
```
Comment (devbww) [inline ci/kokoro/docker/build.sh]: Are these left around on purpose?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -142,13 +142,11 @@ advised that `clang-format` has been known to generate slightly different
 formatting in different versions. We use version 7; use the same version if you
 run into problems.
 
-If you have prepared a Docker image for `ubuntu:18.04` (see below), you can
-verify and fix the format of your code using:
+You canv verify and fix the format of your code using:
```
Comment (remyabel) [inline CONTRIBUTING.md]: s/canv/can

File: ci/kokoro/docker/build.sh [hunk 3]
```diff
@@ -293,11 +293,11 @@ echo "================================================================"
 "${PROJECT_ROOT}/ci/kokoro/docker/upload-coverage.sh" "${docker_flags[@]}"
 
 echo "================================================================"
-#"${PROJECT_ROOT}/ci/travis/dump-logs.sh"
+"${PROJECT_ROOT}/ci/travis/dump-logs.sh"
```
Comment (devbww) [inline ci/kokoro/docker/build.sh]: I assume that "Remove last Travis build" means the days of ci/travis are numbered?

Comment (coryan) [reply ci/kokoro/docker/build.sh]: Yes. Some scripts will get refactored, others just removed.

**Pull request merged**